src/todolistconverters.js

```diff
@@ -0,0 +1,240 @@
+ * @param {module:engine/model/model~Model} model Model instance.
+ * @returns {Function} Returns a conversion callback.
+ */
+export function modelViewChangeType( model ) {
+	return ( evt, data, conversionApi ) => {
+		const viewItem = conversionApi.mapper.toViewElement( data.item );
+		const viewWriter = conversionApi.writer;
+
+		// Add or remove checkbox for toto list.
+		if ( data.attributeNewValue == 'todo' ) {
+			addTodoElementsToListItem( data.item, viewItem, viewWriter, model );
+		} else if ( data.attributeOldValue == 'todo' ) {
+			removeTodoElementsFromListItem( data.item, viewItem, viewWriter, model );
+		}
+	};
+}
+
+/**
+ * @see module:engine/conversion/downcastdispatcher~DowncastDispatcher#event:attribute
+ * @param {module:engine/model/model~Model} model Model instance.
+ * @returns {Function} Returns a conversion callback.
+ */
+export function modelViewChangeChecked( model ) {
+	return ( evt, data, conversionApi ) => {
+		if ( !conversionApi.consumable.consume( data.item, 'attribute:todoListChecked' ) ) {
+			return;
+		}
+
+		const { mapper, writer: viewWriter } = conversionApi;
+		const isChecked = !!data.item.getAttribute( 'todoListChecked' );
+		const viewItem = mapper.toViewElement( data.item );
+		const itemRange = viewWriter.createRangeIn( viewItem );
+		const oldCheckmarkElement = findInRange( itemRange, item => item.is( 'uiElement' ) ? item : false );
+		const newCheckmarkElement = createCheckmarkElement( data.item, viewWriter, isChecked, model );
+
+		viewWriter.insert( viewWriter.createPositionAfter( oldCheckmarkElement ), newCheckmarkElement );
+		viewWriter.remove( oldCheckmarkElement );
+	};
+}
+
+function addTodoElementsToListItem( modelItem, viewItem, viewWriter, model ) {
+	const isChecked = !!modelItem.getAttribute( 'todoListChecked' );
+	const checkmarkElement = createCheckmarkElement( modelItem, viewWriter, isChecked, model );
+
+	viewWriter.addClass( 'todo-list', viewItem.parent );
+	viewWriter.insert( viewWriter.createPositionAt( viewItem, 0 ), checkmarkElement );
+}
+
+function removeTodoElementsFromListItem( modelItem, viewItem, viewWriter, model ) {
+	viewWriter.removeClass( 'todo-list', viewItem.parent );
+	viewWriter.remove( viewItem.getChild( 0 ) );
+	model.change( writer => writer.removeAttribute( 'todoListChecked', modelItem ) );
+}
+
+function createCheckmarkElement( modelItem, viewWriter, isChecked, model ) {
+	const uiElement = viewWriter.createUIElement(
+		'label',
+		{
+			class: 'todo-list__checkmark',
+			contenteditable: false
+		},
+		function( domDocument ) {
+			const checkbox = createElement( document, 'input', { type: 'checkbox', } );
+
+			checkbox.checked = isChecked;
+
+			checkbox.addEventListener( 'change', evt => {
+				model.change( writer => {
+					if ( evt.target.checked ) {
+						writer.setAttribute( 'todoListChecked', true, modelItem );
+					} else {
+						writer.removeAttribute( 'todoListChecked', modelItem );
```

Comment:
As we already talked: this should be a command.

Reply:
Yep. It's on todo list.